civil.go

```diff
@@ -415,6 +423,25 @@ func (dt *DateTime) UnmarshalText(data []byte) error {
 
 // UnmarshalJSON implements encoding/json Unmarshaler interface
 func (dt *DateTime) UnmarshalJSON(data []byte) error {
+	if dt == nil {
```

Comment:
while this check can catch nil DateTime(s), it's not something we need to check for, unmarshal will fail with something like `json: Unmarshal(nil <T>)`

Reply:
@chrsm I added this check due to code on lines
#435: `err := dt.Date.UnmarshalJSON(dataDate)`
#438 `err := dt.Time.UnmarshalJSON(dataTime)`

i.e. `dt` gets dereferences before invoking `UnmarshalJSON`.

While it is *unlikely* that someone will invoke this function on a typed `nil` value, it is possible.

Reply:
It's not about it being unlikely, it's that the nil check will never be used if the pointer is nil, so there's no point to having it.

```go
func TestJsonDec(t *testing.T) {
	x := []byte(`"2020-02-02T23:59:59"`)

	var dt *DateTime

	err := json.Unmarshal(x, dt)
	t.Logf("dt: %#v, err: %s", dt, err)

	var dtp *DateTime
	err = json.Unmarshal(x, &dtp)
	t.Logf("dtp: %#v, err: %s", dtp, err)
}
```

```
$ go test -v -run TestJsonDec
....
    civil_test.go:508: dt: (*civil.DateTime)(nil), err: json: Unmarshal(nil *civil.DateTime)
    civil_test.go:512: dtp: &civil.DateTime{Date:civil.Date{Year:2020, Month:2, Day:2}, Time:civil.Time{Hour:23, Minute:59, Second:59, Nanosecond:0}}, err: %!s(<nil>)
```

Reply:
removed in latest update.